### ISSUE TITLE
flow_loss implementation in HuggingFace may have an issue related to velocity computation.

### ISSUE THREAD

**Issue by liq22** [thuml/Sundial#4]
Hi, thanks for your work.
Should the variable **target** be equal to **target - noise** in your [flow loss](https://huggingface.co/thuml/sundial-base-128m/blob/main/flow_loss.py#L34)? 

This version seems to predict the target sample directly, rather than the velocity.

     def forward(self, target, z, mask=None, mask_y=None):

        noise = torch.randn_like(target)

        t = torch.rand(target.shape[0], device=target.device)

        noised_target = t[:, None] * target + (1 - t[:, None]) * noise

        predict_v = self.net(noised_target, t * 1000, z)

        weights = 1.0 / \
            torch.arange(1, self.in_channels + 1, dtype=torch.float32, device=target.device)
        if mask_y is not None:
            loss = (mask_y * weights * (predict_v - target) ** 2).sum(dim=-1)
        else:
            loss = (weights * (predict_v - target) ** 2).sum(dim=-1)

        if mask is not None:
            loss = (loss * mask).sum() / mask.sum()
        return loss.mean()

**Comment by WenWeiTHU** [comment]
Hi，we have a discussion here： https://huggingface.co/thuml/sundial-base-128m/discussions/2

**Comment by hvlgo** [comment]
You treat `predict_v + noise` as the new `predict_v`, effectively moving the noise addition step inside the model. As a result, during inference, you need to use `pred - noise`. Essentially, it's the same in principle—just two different implementations. The approach used in the code tends to make training more stable.

The main difference between Flow and standard diffusion lies in the way noise is added linearly; predicting the velocity field is not the fundamental distinction.
